bench/julia.cc

```diff
@@ -0,0 +1,155 @@
+
+            image[ny*i + j] = k;
+        }
+    }
+}
+
+template<typename T>
+void julia_v(Scalar<T> xmin, Scalar<T> xmax, size_t nx,
+                  Scalar<T> ymin, Scalar<T> ymax, size_t ny,
+                  Scalar<Index<T>> max_iter, unsigned char *image, Scalar<T> real, Scalar<T> im)
+{
+    T iota;
+    for (size_t i = 0; i < VectorSize<T>(); ++i)
+        Set<T>(iota, i, i);
+
+    T dx = T(xmax - xmin) / T(nx);
+    T dy = T(ymax - ymin) / T(ny), dyv = iota * dy;
+
+    for (size_t i = 0; i < nx; ++i) {
+        for (size_t j = 0; j < ny; j += VectorSize<T>()) {
+            Scalar<Index<T>> k{0};
+            T x = xmin + T(i) * dx,       cr = real, zr = x;
+            T y = ymin + T(j) * dy + dyv, ci = im, zi = y;
+
+            Index<T> kv{0};
+            Mask<T> m{true};
+
+            do {
+                x = zr*zr - zi*zi + cr;
+                y = T(2.0) * zr*zi + ci;
+                MaskedAssign<T>(zr, m, x);
+                MaskedAssign<T>(zi, m, y);
+                MaskedAssign<Index<T>>(kv, m, ++k);
+                m = zr*zr + zi*zi < T(4.0);
+            } while (k < max_iter && !MaskEmpty(m));
+
+            for (size_t k = 0; k < VectorSize<T>(); ++k)
+                image[ny*i + j + k] = (unsigned char) Get(kv, k);
+        }
+    }
+}
+
+template<typename T>
+void bench_julia(T xmin, T xmax, size_t nx, T ymin, T ymax, size_t ny,
+                      int max_iter, unsigned char *image, const char *backend, T cr, T ci)
+{
+    std::string filename = "julia_" + std::string(backend) + ".png";
+    Timer<milliseconds> timer;
+    julia<T>(xmin, xmax, nx, ymin, ymax, ny, max_iter, image, cr, ci);
+    printf("%15s: %7.2lf ms\n", backend, timer.Elapsed());
+    write_png(filename.c_str(), image, nx, ny);
+}
+
+template<typename T>
+void bench_julia_v(Scalar<T> xmin, Scalar<T> xmax, size_t nx,
+                        Scalar<T> ymin, Scalar<T> ymax, size_t ny,
+                        int max_iter, unsigned char *image, const char *backend, Scalar<T> cr, Scalar<T> ci)
+{
+    std::string filename = "julia_" + std::string(backend) + ".png";
+    Timer<milliseconds> timer;
+    julia_v<T>(xmin, xmax, nx, ymin, ymax, ny, max_iter, image, cr, ci);
+    printf("%15s: %7.2lf ms\n", backend, timer.Elapsed());
+    write_png(filename.c_str(), image, nx, ny);
+}
+
+int main(int argc, char *argv[])
+{
+    double xmin = -2, xmax = 2;
+    double ymin = -2, ymax = 2;
+
+    size_t nx = 2400, ny = 2025, max_iter = 200;
+    double cr = 0.285, ci = 0.01;
+    std::cout<<cr<<std::endl;    
+    std::cout<<ci<<std::endl;    
+    unsigned char *image = new unsigned char[nx*ny];
+
+   if(argc>1) {
+      if(argc!=3) {
+        std::cerr<<"Error. Incorrect number of parameters for the c parameter in the quadratic polinomial z\x00B2+c. Usage: julia [<realComponent> <imaginary component>] "<<std::endl;
+        exit(-1);
+      } else {
+        cr = atof(argv[1]);
+        ci = atof(argv[2]);
+      }
+   }
```

Comment:
Please format like the rest of the code (4 spaces rather than 3, and spaces after `if`, etc).